source/core/TypeAndDataManager.cpp

```diff
@@ -2453,8 +2453,7 @@ Boolean TypedArrayCore::ResizeDimensions(Int32 rank, IntIndex *dimensionLengths,
     }
 
     // 3. If more actual elements are needed, initialize the new ones (or all of them if requested)
-    // TODO(PaulAustin): honor bOK status.
-    if (!preserveElements) {
+    if (!preserveElements && bOK) {
         ElementType()->InitData(BeginAt(0), newLength);
```

Comment:
If we set the bOK to false then we already have some issue so added a check. ([This ](https://github.com/ni/VireoSDK/blob/c8eb11d197b09f7ed2100bdb01cd48cd8f2fa003/test-it/karma/publicapi/ExecuteSlicesUntilClumpsFinshed.Test.js#L118)test failing due to this )

Reply:
The existing vireo  [test](https://github.com/ni/VireoSDK/blob/c8eb11d197b09f7ed2100bdb01cd48cd8f2fa003/test-it/karma/publicapi/ExecuteSlicesUntilClumpsFinshed.Test.js#L118) was failing after emsdk update.
I think the intention of this test to check the overflow error report generated [here](https://github.com/ni/VireoSDK/blob/c8eb11d197b09f7ed2100bdb01cd48cd8f2fa003/source/core/TypeAndDataManager.cpp#L162). But bOK (which suppose to set to false on the error)  flag does not tested down the code and trying to allocate/access the wrong location which leads to a run time segmentation fault. So added a bOK check before accessing the location. 
For this particular test, we already know about the error before this line, and bOK is set to false so we don't want to do `InitData` which leads to a segmentation fault.

Reply:
Discussed in person and instead decided failed allocations should abort the runtime